internal/api/v1/user/merkleroots/merkleroots_api.go

```diff
@@ -0,0 +1,49 @@
+package merkleroots
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/user/querybuilders"
+	"github.com/bitcoin-sv/spv-wallet-go-client/queries"
+	"github.com/go-resty/resty/v2"
+)
+
+const route = "api/v1/merkleroots"
+
+type API struct {
+	addr       string
+	httpClient *resty.Client
+}
+
+func (a *API) MerkleRoots(ctx context.Context, merkleRootOpts ...queries.MerkleRootsQueryOption) (*queries.MerkleRootPage, error) {
+	var query queries.MerkleRootsQuery
+	for _, o := range merkleRootOpts {
+		o(&query)
+	}
+
+	queryBuilder := querybuilders.NewQueryBuilder(querybuilders.WithFilterQueryBuilder(&merkleRootsFilterQueryBuilder{query: query}))
+	params, err := queryBuilder.Build()
+	if err != nil {
+		return nil, fmt.Errorf("failed to build merkle roots query params: %w", err)
+	}
+
+	var result queries.MerkleRootPage
+	_, err = a.httpClient.R().
+		SetContext(ctx).
+		SetResult(&result).
+		SetQueryParams(params.ParseToMap()).
+		Get(a.addr)
+	if err != nil {
+		return nil, fmt.Errorf("HTTP response failure: %w", err)
+	}
+
+	return &result, nil
+}
+
+func NewAPI(addr string, httpClient *resty.Client) *API {
+	return &API{
+		addr:       addr + "/" + route,
```

Comment:
Use `path.Join` to handle slashes properly.
```suggestion
		addr:       path.Join(addr, route),
```

Reply:
I created dedicated task: SPV-1209 to handle the issue. This scheme is repeated across another packages and should be fixed asap.